### PR TITLE
🎨♻️Use structured parent project/node in director-v2 and connect with RUT

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
Follow-up on https://github.com/ITISFoundation/osparc-simcore/pull/5874

This upgrades the director-v2 to directly take the parent project/node from the ```projects_metadata``` table for consistency.

## Related issue/s
- requires https://github.com/ITISFoundation/osparc-simcore/pull/5874
- requires https://github.com/ITISFoundation/osparc-simcore/issues/5878

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/director-v2/src/simcore_service_director_v2/api/routes/computations.py [hunk 1]
```diff
@@ -155,39 +147,64 @@ async def _check_pipeline_startable(
...
-        ProjectNodesNodeNotFoundError,
-        ProjectNodesNonUniqueNodeFoundError,
-    ):
-        _logger.exception("Could not find project/node: %s", f"{parent_node_id=}")
-        return {}
+
+    except DBProjectNotFoundError:
+        _logger.exception("Could not find project: %s", f"{project_id=}")
+    except ProjectNotFoundError as exc:
+        _logger.exception("Could not find parent project: %s", f"{exc.project_id=}")
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/api/routes/computations.py:205]: Q: this does not print the traceback right? If it does not print it, all good.

**Review by GitHK — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/api/routes/computations.py (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/api/routes/computations.py:205]: why? yes it does. what is the problem with this?

File: services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py [hunk 2]
```diff
@@ -106,11 +111,11 @@ async def publish_service_resource_tracking_started(  # pylint: disable=too-many
...
-        parent_project_id=project_id,  # <-- SAN please modify
-        root_parent_project_id=project_id,  # <-- SAN please modify
-        root_parent_project_name=project_name,  # <-- SAN please modify
-        parent_node_id=node_id,  # <-- SAN please modify
-        root_parent_node_id=node_id,  # <-- SAN please modify
+        parent_project_id=parent_project_id or project_id,
+        root_parent_project_id=root_parent_project_id or project_id,
+        root_parent_project_name=root_parent_project_name or project_name,
+        parent_node_id=parent_node_id or node_id,
+        root_parent_node_id=root_parent_node_id or node_id,
```
Comment (matusdrobuliak66) [inline services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py:118]: Super thanks! And yes if you think this logic should be moved to RUT I can move it there afterwards.

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: services/director-v2/src/simcore_service_director_v2/api/routes/computations.py [hunk 3]
```diff
@@ -155,39 +147,64 @@ async def _check_pipeline_startable(
...
-    except (
-        ProjectNotFoundError,
-        ProjectNodesNodeNotFoundError,
-        ProjectNodesNonUniqueNodeFoundError,
-    ):
-        _logger.exception("Could not find project/node: %s", f"{parent_node_id=}")
-        return {}
+
+    except DBProjectNotFoundError:
+        _logger.exception("Could not find project: %s", f"{project_id=}")
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/api/routes/computations.py:203]: TIP: When exceptions are well defined (e.g. inherit from `OsparcErrorMixin`, well defined msg_template, etc ) this handlers would simplify further as 
```python
try
   #...
except (DBProjectNotFoundError,ProjectNotFoundError ) as err:
        _logger.exception("Fatal error %s. See further details for diagnostics", err.code, extras=...)
```

**Review by pcrespov — APPROVED** [review]
thx

File: services/director-v2/src/simcore_service_director_v2/models/comp_runs.py [hunk 4]
```diff
@@ -18,6 +18,10 @@ class ProjectMetadataDict(TypedDict, total=False):
     parent_node_name: str
     parent_project_id: ProjectID
     parent_project_name: str
+    root_parent_project_id: ProjectID
+    root_parent_project_name: str
+    root_parent_node_id: NodeID
+    root_parent_node_name: str
```
Comment (bisgaard-itis) [inline services/director-v2/src/simcore_service_director_v2/models/comp_runs.py:24]: Question: Are these also supposed to be picked up by the `osparc` client from the environment and propagated through the api-server?

**Review by bisgaard-itis — APPROVED** [review]
Cool! Thanks a lot 👍🏻

File: services/director-v2/src/simcore_service_director_v2/models/comp_runs.py (hunk 4 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/models/comp_runs.py:24]: no. I take care of this.

File: services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py (hunk 2 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/utils/rabbitmq.py:118]: discussed. I let it be for now.

File: services/director-v2/src/simcore_service_director_v2/api/routes/computations.py (hunk 3 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/api/routes/computations.py:203]: alright. will try to remember that OsparcErrorMixin for next times

**Pull request merged**